src/script/components/Conversation/ConversationCells/CellsHeader/CellsNewMenu/CellsNewMenu.tsx

```diff
@@ -0,0 +1,95 @@
+/*
+ * Wire
+ * Copyright (C) 2025 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {KeyboardEvent, MouseEvent as ReactMouseEvent, useState} from 'react';
+
+import {QualifiedId} from '@wireapp/api-client/lib/user';
+
+import {Button, ButtonVariant, PlusIcon} from '@wireapp/react-ui-kit';
+
+import {CellsRepository} from 'src/script/cells/CellsRepository';
+import {showContextMenu} from 'src/script/ui/ContextMenu';
+import {isSpaceOrEnterKey} from 'Util/KeyboardUtil';
+import {t} from 'Util/LocalizerUtil';
+import {setContextMenuPosition} from 'Util/util';
+
+import {CellsNewItemModal} from './CellsNewItemModal/CellsNewItemModal';
+import {buttonStyles, iconStyles} from './CellsNewMenu.styles';
+
+import {CellItem} from '../../common/cellFile/cellFile';
+
+interface CellsNewMenuProps {
+  cellsRepository: CellsRepository;
+  conversationQualifiedId: QualifiedId;
+  onRefresh: () => void;
+}
+
+export const CellsNewMenu = ({cellsRepository, conversationQualifiedId, onRefresh}: CellsNewMenuProps) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [modalType, setModalType] = useState<CellItem['type']>('file');
+
+  const showOptionsMenu = (event: ReactMouseEvent<HTMLButtonElement> | MouseEvent) => {
+    showContextMenu({
+      event,
+      entries: [
+        {
+          label: t('cellsNewItemMenu.folder'),
+          click: () => {
+            setModalType('folder');
+            setIsModalOpen(true);
+          },
```

Comment:
I think You can make a separate function
openModal(modalType: 'folder' | 'file') {...} and reuse it on line 60 ;)
```suggestion
          click: () => openModal('folder'),
```